python/lsst/daf/persistence/posixStorage.py

```diff
@@ -615,5 +540,248 @@ def storageExists(uri):
+    if pythonType is not None:
+        if isinstance(pythonType, basestring):
+            pythonType = doImport(pythonType)
+    results = []
+    additionalData = butlerLocation.getAdditionalData()
+    for locationString in butlerLocation.getLocations():
+        locStringWithRoot = os.path.join(butlerLocation.getStorage().root, locationString)
+        logLoc = LogicalLocation(locStringWithRoot, additionalData)
+        if not os.path.exists(logLoc.locString()):
+            raise RuntimeError("No such FITS catalog file: " + logLoc.locString())
+        INT_MIN = -(1 << 31)
+        hdu = additionalData.getInt("hdu", INT_MIN)
+        flags = additionalData.getInt("flags", 0)
+        finalItem = pythonType.readFits(logLoc.locString(), hdu, flags)
+        results.append(finalItem)
+    return results
+
+
+def writeFitsCatalogStorage(butlerLocation, obj):
+    """Writes an object to a location and persistence format specified by
+    ButlerLocation
+
+    Parameters
+    ----------
+    butlerLocation : ButlerLocation
+        The location & formatting for the object to be written.
+    obj : object instance
+        The object to be written.
+    """
+    additionalData = butlerLocation.getAdditionalData()
+    locations = butlerLocation.getLocations()
+    with SafeFilename(os.path.join(butlerLocation.getStorage().root, locations[0])) as locationString:
+        logLoc = LogicalLocation(locationString, additionalData)
+        flags = additionalData.getInt("flags", 0)
+        obj.writeFits(logLoc.locString(), flags=flags)
+        return
+
+
+def readPafStorage(self, butlerLocation):
+    """Read from a butlerLocation.
+
+    Parameters
+    ----------
+    butlerLocation : ButlerLocation
+        The location & formatting for the object(s) to be read.
+
+    Returns
+    -------
+    A list of objects as described by the butler location. One item for
+    each location in butlerLocation.getLocations()
+    """
+    results = []
+    for locationString in butlerLocation.getLocations():
+        logLoc = LogicalLocation(butlerLocation.getStorage().locationWithRoot(locationString),
+                                 butlerLocation.getAdditionalData())
+        finalItem = pexPolicy.Policy.createPolicy(logLoc.locString())
+        results.append(finalItem)
+    return results
+
+
+def readYamlStorage(butlerLocation):
+    """Read from a butlerLocation.
+
+    Parameters
+    ----------
+    butlerLocation : ButlerLocation
+        The location & formatting for the object(s) to be read.
+
+    Returns
+    -------
+    A list of objects as described by the butler location. One item for
+    each location in butlerLocation.getLocations()
+    """
+    results = []
+    for locationString in butlerLocation.getLocations():
+        logLoc = LogicalLocation(butlerLocation.getStorage().locationWithRoot(locationString),
+                                 butlerLocation.getAdditionalData())
+        finalItem = Policy(filePath=logLoc.locString())
```

Comment:
I guess this Policy is different from pexPolicy? Sort of confusing.

Reply:
yeah. naming is hard. sorry.